### PR TITLE
Use Kitura-WebSocket-Compression for WebSocket Compression Handlers 

### PR DESCRIPTION
**Pull request by harish1992**
Previously `Kitura-WebSocket-NIO` used to add local compression handlers which only supported `PermessageDeflate` algorithm for compression.

This PR removes `PermessageDeflateCompresser` and `PermessageDeflateCompresser` from the library and uses the compression handlers from new `Kitura-WebSocket-Compression` library for and when the compression is negotiated over a WebSocket connection.

### CODE REVIEW

File: Tests/KituraWebSocketTests/WebSocketClient.swift [hunk 1]
```diff
@@ -40,12 +41,15 @@ class WebSocketClient {
     public var maxWindowBits: Int32
     public var contextTakeover: ContextTakeover
     public var maxFrameSize: Int
+//    var sslEnabled: Bool = false
```
Comment (pushkarnk) [inline Tests/KituraWebSocketTests/WebSocketClient.swift]: Please delete all commented code.

File: Tests/KituraWebSocketTests/WebSocketClient.swift [hunk 2]
```diff
@@ -431,7 +519,7 @@ class WebSocketMessageHandler: ChannelInboundHandler, RemovableChannelHandler {
 class HTTPClientHandler: ChannelInboundHandler, RemovableChannelHandler {
 
     typealias InboundIn = HTTPClientResponsePart
-    unowned var client: WebSocketClient
```
Comment (pushkarnk) [inline Tests/KituraWebSocketTests/WebSocketClient.swift]: @harish1992 Are you sure we don't need this to be an `unowned` reference any more?

File: Tests/KituraWebSocketTests/WebSocketClient.swift (hunk 1 above)
Comment (harish1992) [reply Tests/KituraWebSocketTests/WebSocketClient.swift]: Done.

File: Tests/KituraWebSocketTests/WebSocketClient.swift (hunk 2 above)
Comment (harish1992) [reply Tests/KituraWebSocketTests/WebSocketClient.swift]: The reference is required. Thanks for pointing out.

**Pull request merged**